hermit-sys/build.rs

```diff
@@ -80,6 +81,55 @@ fn build_hermit(src_dir: &Path, target_dir_opt: Option<&Path>) {
 	} else {
 		panic!("Try to build for an unsupported platform");
 	};
+	println!("Lib location: {}", lib_location.display());
+
+	if profile == "debug" {
+		// get access to llvm tools shipped in the llvm-tools-preview rustup component
+		let llvm_tools = match llvm_tools::LlvmTools::new() {
+			Ok(tools) => tools,
+			Err(llvm_tools::Error::NotFound) => {
+				eprintln!("Error: llvm-tools not found");
+				eprintln!("Maybe the rustup component `llvm-tools-preview` is missing?");
+				eprintln!("  Install it through: `rustup component add llvm-tools-preview`");
+				process::exit(1);
+			}
+			Err(err) => {
+				eprintln!("Failed to retrieve llvm-tools component: {:?}", err);
+				process::exit(1);
+			}
+		};
+
+		let lib = lib_location.join("libhermit.a");
+
+		// determine llvm_objcopy
+		let llvm_objcopy = llvm_tools
+			.tool(&llvm_tools::exe("llvm-objcopy"))
+			.expect("llvm_objcopy not found in llvm-tools");
+
+		// rename symbols
+		let mut cmd = Command::new(llvm_objcopy);
+		cmd.arg("--redefine-sym")
+			.arg("memcpy=kernel_memcpy")
+			.arg("--redefine-sym")
+			.arg("memmove=kernel_memmove")
+			.arg("--redefine-sym")
+			.arg("memset=kernel_memset")
+			.arg("--redefine-sym")
+			.arg("memcmp=kernel_memcmp")
+			.arg("--redefine-sym")
+			.arg("bcmp=kernel_bcmp")
+			.arg(lib.display().to_string());
+
+		println!("cmd {:?}", cmd);
+		let output = cmd.output().expect("Unable to rename symbols");
+		let stdout = std::string::String::from_utf8(output.stdout);
+		let stderr = std::string::String::from_utf8(output.stderr);
+		println!("Rename symbols output-status: {}", output.status);
+		println!("Rename symbols output-stdout: {}", stdout.unwrap());
+		println!("Rename symbols output-stderr: {}", stderr.unwrap());
+		assert!(output.status.success());
+	}
```

Comment:
Can you add a comment explaining what you are doing and why you are doing this?

Reply:
Just added (see below)